src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ReaderService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+public class ReaderServiceImpl implements ReaderService {
+    @Override
+    public List<FruitTransaction> readFromFile(String filePath) {
```

Comment:
split the logic of this method into two services. one will read strings from the file and the other will parse these strings into FruitTransaction. with such services will follow SRP and can be used independently